src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -273,148 +395,76 @@ public boolean isToolInHut(final String tool, @NotNull final AbstractBuilding re
      * It will be taken from the chest and placed in the workers inventory.
      * Make sure that the worker stands next the chest to not break immersion.
      * Also make sure to have inventory space for the stack.
-     *
      * @param entity the tileEntity chest or building.
-     * @param is     the itemStack.
+     * @param itemStackSelectionPredicate the itemStack predicate.
      * @return true if found the stack.
      */
-    public boolean isInTileEntity(TileEntityChest entity, ItemStack is)
+    private boolean isInTileEntity(final TileEntityChest entity, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
     {
-        return is != null
-                 && InventoryFunctions
-                      .matchFirstInProviderWithAction(
+        return InventoryFunctions
+                .matchFirstInProviderWithAction(
                         entity,
-                        stack -> !InventoryUtils.isItemStackEmpty(stack) && is.isItemEqual(stack),
-                        InventoryFunctions::doNothing
-                      );
+                        itemStackSelectionPredicate,
+                        InventoryFunctions::doNothing);
     }
 
-    /**
-     * Get the first task in the list.
-     *
-     * @return the building which needs a delivery.
-     */
-    @Nullable
-    public AbstractBuilding getTask()
-    {
-        if (list.isEmpty())
-        {
-            return null;
-        }
-        return list.remove(0);
-    }
-
-    /**
-     * Check for a certain item and return the position of the chest containing it.
-     *
-     * @param stack the stack to search for.
-     * @return the position or null.
-     */
-    @Nullable
-    public BlockPos getPositionOfChestWithItemStack(ItemStack stack)
+    @Override
+    public void readFromNBT(final NBTTagCompound compound)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method to simply inherit it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)